### PR TITLE
New Challenges:  Information Retrieval for SpaceX, Anthropic, AutoGPT, Milvius

### PR DESCRIPTION
**Pull request by rihp**
### Background

This is an information retrieval challenge to get AutoGPT to find new information that is not known already by the LLM

### Changes

only the files for the challenge

### Test Plan

`pytest tests/integration/challenges/information_retrieval/test_information_retrieval_challenge_b.py -v  `

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by rihp** [comment]
Related to #4244

File: tests/integration/agent_factory.py [hunk 1]
```diff
@@ -216,3 +216,34 @@ def kubernetes_agent(memory_local_cache, workspace: Workspace):
...
+        ai_name="Anthropic-Information",
+        ai_role="an autonomous agent that specializes in researching about Anthropic's new LLM.",
+        ai_goals=[
+            "Write in a file called output.txt how many tokens fit in the Anthropic LLM which was announced in Q2 2023",
+        ],
+    )
+    ai_config.command_registry = command_registry
+
+    system_prompt = ai_config.construct_full_prompt()
+    Config().set_continuous_mode(False)
```
Comment (merwanehamadi) [inline tests/integration/agent_factory.py:236]: use agent_test_config

File: tests/integration/agent_factory.py [hunk 2]
```diff
@@ -216,3 +216,34 @@ def kubernetes_agent(memory_local_cache, workspace: Workspace):
     )
 
     return agent
+
```
Comment (merwanehamadi) [inline tests/integration/agent_factory.py:219]: modify the docs please

**Review by merwanehamadi — APPROVED** [review]

File: tests/integration/agent_factory.py [hunk 3]
```diff
@@ -216,3 +216,34 @@ def kubernetes_agent(memory_local_cache, workspace: Workspace):
...
+@pytest.fixture
+def anthropic_information_agent(memory_local_cache, workspace: Workspace):
```
Comment (waynehamadi) [inline tests/integration/agent_factory.py]: modify the docs folder to add this challenge please please

File: tests/integration/agent_factory.py (hunk 1 above)
Comment (waynehamadi) [inline tests/integration/agent_factory.py:239]: use agent_test_config please

**Review by merwanehamadi — CHANGES_REQUESTED** [review]

File: tests/integration/challenges/information_retrieval/test_information_retrieval_challenge_b.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+from tests.integration.challenges.utils import run_interaction_loop, run_multiple_times
+from tests.utils import requires_api_key
+
+CYCLE_COUNT = 3
+from autogpt.agent import Agent
+
+
+# @pytest.mark.skip("This challenge hasn't been beaten yet.")
+@pytest.mark.vcr
+@requires_api_key("OPENAI_API_KEY")
```
Comment (waynehamadi) [inline tests/integration/challenges/information_retrieval/test_information_retrieval_challenge_b.py:14]: skip the test I don't think the challenge is beaten

**Review by merwanehamadi — CHANGES_REQUESTED** [review]

**Comment by piotrmasior** [comment]
> This challenge passes locally

I am not DevOps expert for github or your stack but resolving that file /tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace/arxiv_paper.txt looks like a problem with some chmod or disk size or whatever

it is visible at:

```
023-05-16T20:28:22.0002130Z [33mCRITICISM: [0m I should ensure that the paper I am reading is the correct one and that I am searching for the correct information.
2023-05-16T20:28:22.0002583Z  [0m Resolving path 'arxiv_paper.txt' in workspace '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace'
2023-05-16T20:28:22.0002965Z  [0m Resolved root as '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace'
2023-05-16T20:28:22.0003381Z  [0m Joined paths as '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace/arxiv_paper.txt'
2023-05-16T20:28:22.0003918Z [36mNEXT ACTION: [0m COMMAND = [36mread_file[0m ARGUMENTS = [36m{'filename': '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace/arxiv_paper.txt'}[0m
2023-05-16T20:28:22.0004355Z  [0m Enter 'y' to authorise command, 'y -N' to run N continuous commands, 's' to run self-feedback commands, 'n' to exit program, or enter feedback for Twitter-Demo...
2023-05-16T20:28:22.0004610Z  [0m Asking user via keyboard...
2023-05-16T20:28:22.0004827Z [35m-=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= [0m
2023-05-16T20:28:22.0005372Z [33mSYSTEM: [0m Command read_file returned: Error: [Errno 2] No such file or directory: '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace/arxiv_paper.txt'
2023-05-16T20:28:22.0005488Z - Thinking... 
```
and al lot of thinkins is going later on...

for the gpt error it just couldnt resolve alternative path -as far I understand it of course

so, test just stuck in the loop because not getting that file is an edge case for test infrastructure, or I am really not understanding this tests... (looks like twitter tests just get around this problem by reaching information from elsewhere), but again - If i am talking just BS I would love to hear someone other idea
best

**Comment by piotrmasior** [comment]
> > This challenge passes locally
> 
> I am not DevOps expert for github or your stack but resolving that file /tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace/arxiv_paper.txt looks like a problem with some chmod or disk size or whatever
> 
> it is visible at:
> 
> ```
> 023-05-16T20:28:22.0002130Z �[33mCRITICISM: �[0m I should ensure that the paper I am reading is the correct one and that I am searching for the correct information.
> 2023-05-16T20:28:22.0002583Z  �[0m Resolving path 'arxiv_paper.txt' in workspace '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace'
> 2023-05-16T20:28:22.0002965Z  �[0m Resolved root as '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace'
> 2023-05-16T20:28:22.0003381Z  �[0m Joined paths as '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace/arxiv_paper.txt'
> 2023-05-16T20:28:22.0003918Z �[36mNEXT ACTION: �[0m COMMAND = �[36mread_file�[0m ARGUMENTS = �[36m{'filename': '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace/arxiv_paper.txt'}�[0m
> 2023-05-16T20:28:22.0004355Z  �[0m Enter 'y' to authorise command, 'y -N' to run N continuous commands, 's' to run self-feedback commands, 'n' to exit program, or enter feedback for Twitter-Demo...
> 2023-05-16T20:28:22.0004610Z  �[0m Asking user via keyboard...
> 2023-05-16T20:28:22.0004827Z �[35m-=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= �[0m
> 2023-05-16T20:28:22.0005372Z �[33mSYSTEM: �[0m Command read_file returned: Error: [Errno 2] No such file or directory: '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_information_retrieval_cha5/home/users/monty/auto_gpt_workspace/arxiv_paper.txt'
> 2023-05-16T20:28:22.0005488Z - Thinking... 
> ```
> 
> and a lot of 'Thinking's is going later on...
> 
> for the gpt error it just couldnt resolve alternative 'thinking' path - as far I understand it of course.
> 
> so, test just stuck in the loop because not getting that file is an edge case for test infrastructure, or I am really not understanding this tests... (looks like twitter tests just get around this problem by reaching information from elsewhere), but again - If i am talking just BS I would love to hear someone other idea best

also lines before...

```
2023-05-16T20:28:22.0000627Z [33mTWITTER-DEMO THOUGHTS: [0m I will use the read_file command to read the arXiv paper and search for the name of the main author.
2023-05-16T20:28:22.0001001Z [33mREASONING: [0m The arXiv paper is likely to contain the information I need, and the read_file command will allow me to access it.
2023-05-16T20:28:22.0001118Z [33mPLAN: [0m
2023-05-16T20:28:22.0001335Z [32m- [0m Use the read_file command to access the arXiv paper
2023-05-16T20:28:22.0001550Z [32m- [0m Search the paper for the name of the main author
2023-05-16T20:28:22.0001755Z [32m- [0m Save the name of the main author to output.txt
```

all examples are with output.txt what happened during that text and it was changed to 'arxiv_paper.txt'

**Comment by piotrmasior** [comment]
> This challenge passes locally

```
an autonomous agent that specializes in researching and saving files as output.txt 
```
is all in agent_factory and I am not fully sure but I have intuition it has something with recent changes in memory handlers, so happy debugging xD

**Comment by piotrmasior** [comment]
> > This challenge passes locally

`"content": "{\n    \"thoughts\": {\n        \"text\": \"I will use the read_file command to read the arXiv paper and search for the name of the main author.\",\n        \"reasoning\": \"The arXiv paper is likely to contain the information I need, and the read_file command will allow me to access it.\",\n        \"plan\": \"- Use the read_file command to access the arXiv paper\\n- Search the paper for the name of the main author\\n- Save the name of the main author to output.txt\",\n        \"criticism\": \"I should ensure that the paper I am reading is the correct one and that I am searching for the correct information.\",\n        \"speak\": \"I will use the read_file command to access the arXiv paper and search for the name of the main author.\"\n    },\n    \"command\": {\n        \"name\": \"read_file\",\n        \"args\": {\n            \"filename\": \"arxiv_paper.txt\"\n        }\n    }\n}"`

**Comment by rihp** [comment]
Updated challenges and marked to skip the ones that arent passing consistently

**Comment by rihp** [comment]
updated to include only the two tests that pass more consistenty

**Comment by rihp** [comment]
> > updated to include only the two tests that pass more consistenty
> 
> at this point my thoughts are only, wtf is with the test infrastructure
sure! what questions do you have more specifically ?

**Comment by piotrmasior** [comment]
> > > updated to include only the two tests that pass more consistenty
> > 
> > 
> > at this point my thoughts are only, wtf is with the test infrastructure
> 
> sure! what questions do you have more specifically ?

Hey, I'm glad you asked. First, I see that you're struggling with a problem related to files, which was the main focus of my previous question. I believe it might be tied to the functioning of the test infrastructure, but I lack complete insights here.

However, since we're already in a discussion, I'd like to present my second point as both an opinion and a question. Why do integration tests validate that AutoGPT works when it's written in such a static way, still relying on bypasses and cycles? From what I can see in the code, you're manipulating the number of these cycles in hopes of merely passing the test for its own sake. But doesn't this approach fall short when tested on a larger scale? It seems you'd need to infinitely increase these cycles when creating more tests. Isn't this counterproductive?

**Comment by merwanehamadi** [comment]
> > > > updated to include only the two tests that pass more consistenty
> > > 
> > > 
> > > at this point my thoughts are only, wtf is with the test infrastructure
> > 
> > 
> > sure! what questions do you have more specifically ?
> 
> Hey, I'm glad you asked. First, I see that you're struggling with a problem related to files, which was the main focus of my previous question. I believe it might be tied to the functioning of the test infrastructure, but I lack complete insights here.
> 
> However, since we're already in a discussion, I'd like to present my second point as both an opinion and a question. Why do integration tests validate that AutoGPT works when it's written in such a static way, still relying on bypasses and cycles? From what I can see in the code, you're manipulating the number of these cycles in hopes of merely passing the test for its own sake. But doesn't this approach fall short when tested on a larger scale? It seems you'd need to infinitely increase these cycles when creating more tests. Isn't this counterproductive?

@piotrmasior join us on discord so we talk about it https://discord.gg/autogpt

```still relying on bypasses and cycles```
yeah I don't like this bypass, we will change it to a budget OR define the number of cycles when initializing the agent.
 infinitely increase these cycles => right now each test can be done in less than 5 cycles on average, so for now we don't have this problem, but soon we will
